src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -272,23 +303,20 @@ public HttpResponse execute(HttpUriRequest request, HttpContext context) throws
 
     @Override
     public HttpResponse execute(HttpHost target, HttpRequest request, HttpContext context) throws IOException {
-        if (context == null) {
-            context = this.context;
-        }
-        final ReadWriteLock rwLock = (ReadWriteLock) context.getAttribute(LOCK_RW);
-        Lock readLock = null;
-        if (rwLock != null) {
-            readLock = rwLock.readLock();
-            readLock.lock();
+        final Lock readLock = rwLock.readLock();
+        readLock.lock();
+
+        if (tt != null) {
+            // this adds TT header to EVERY request to ALL hosts made by this HTTP client
```

Comment:
is there some chance to check if the `target` or `request` goes to gooddata host (the same as `this.authHost`)? If not please add the reason to the comment.